services/marathon/marathon.go

```diff
@@ -158,7 +180,7 @@ func createApps(tasksById map[string][]MarathonTask, marathonApps map[string]Mar
 
 		for _, task := range tasks {
 			if len(task.Ports) > 0 {
-				simpleTasks = append(simpleTasks, Task{Host: task.Host, Port: task.Ports[0]})
+				simpleTasks = append(simpleTasks, Task{Host: task.Host, Port: task.Ports[0], Alive: calculateTaskHealth(task.HealthCheckResults, marathonApps[appId].HealthChecks)})
```

Comment:
Will that risk introducing delay in the event that applications go unhealthy?  You'd need to wait for marathon's health checks to fail, send out an event on the bus, and for bamboo to react to it instead of just letting haproxy detect it.  (it's possible that it won't be any worse, I'm just not familiar enough to know).